### PR TITLE
HIVE-26809: Upgrade ORC to 1.8.3

### PR DESCRIPTION
**Pull request by zratkai**
### What changes were proposed in this pull request?
 Upgrade ORC dependency to 1.8.3.

### Why are the changes needed?
ORC version was old, newer version contains many new features like SARG filters.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By running the tests in the pipeline.

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java [hunk 1]
```diff
@@ -298,11 +300,11 @@ public void seek(PositionProvider index) throws IOException {
       }
     }
 
-    @Override
+    @Override        
     public void nextVector(
-        ColumnVector previousVector, boolean[] isNull, int batchSize) throws IOException {
+        ColumnVector previousVector, boolean[] isNull, int batchSize, FilterContext filterContext, TypeReader.ReadPhase readPhase) throws IOException {
```
Comment (abstractdog) [inline ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java:305]: this line looks a bit long :) can you reformat according to [this](https://github.com/apache/hive/blob/master/dev-support/eclipse-styles.xml)?

File: ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java [hunk 2]
```diff
@@ -298,11 +300,11 @@ public void seek(PositionProvider index) throws IOException {
       }
     }
 
-    @Override
+    @Override
```
Comment (abstractdog) [inline ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java:303]: please remove whitespace

File: ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java [hunk 3]
```diff
@@ -462,10 +464,9 @@ public void seek(PositionProvider index) throws IOException {
     }
 
     @Override
-    public void nextVector(
-        ColumnVector previousVector, boolean[] isNull, int batchSize) throws IOException {
+    public void nextVector(ColumnVector previousVector, boolean[] isNull, final int batchSize, FilterContext filterContext, ReadPhase readPhase) throws IOException {
```
Comment (abstractdog) [inline ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java:467]: long line

**Comment by zratkai** [comment]
@abstractdog thanks for the review!
The issue discussed in the other PR was an ORC related issue, so I fixed it in ORC:
https://github.com/apache/orc/pull/1431

File: ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java (hunk 1 above)
Comment (zratkai) [reply ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java]: Reformatted to match the style.

File: ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java (hunk 2 above)
Comment (zratkai) [reply ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java]: Reformatted to match the style.

**Comment by TuroczyX** [comment]
@difin @akshat0395  pls also check this pr.

This change is one of the most interesting change since the Spotify is invented! 

cc: @nrg4878

**Comment by TuroczyX** [comment]
As I see this change mainly contains the new ORC file sizes changes in the tests, and minor code modifications on the hive itself. I do not see to many critical changes here.

**Comment by zhangbutao** [comment]
I am thinking if we can mask or remove the ORC file sizes related qtest items. In this pr, It is so wasting time to fix qtests about orc file size when upgrading orc version. Also, other OSS eg. Spark seems not treat orc file size as a test item.

**Comment by TuroczyX** [comment]
> 

I think in a long term we should. Now it is fixed is fine. But you right pretty over checked. I guess we can create a following ticket. Because this sizes check does not protect the code, and It does not create to much value. Also with new ORC's the files would be always different and check manually is just beh. 
For the first upgrade I would be OK with these if it is already there. (The last upgrade was in 1972 :)  Finally the Hive will have the latest one)

**Comment by zhangbutao** [comment]
Sure, i am also ok with this change. Maybe we can create a following ticket to mask orc file size, then we can easily upgrade orc version next time.

File: ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestOrcFile.java [hunk 4]
```diff
@@ -1234,7 +1234,7 @@ public void emptyFile() throws Exception {
     assertEquals(false, reader.rows().hasNext());
     assertEquals(CompressionKind.NONE, reader.getCompression());
     assertEquals(0, reader.getNumberOfRows());
-    assertEquals(0, reader.getCompressionSize());
+    assertEquals(262144, reader.getCompressionSize());
```
Comment (abstractdog) [inline ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestOrcFile.java:1237]: this looks a bit weird...I mean any non-zero to non-zero change is okay as file size changes, but 0 -> 262144 is strange, do we know the root cause? was the assertion of =0 made intentionally?

Comment (zratkai) [reply ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestOrcFile.java:1237]: I agree it looks weird. The reason is this:
ORC-1357: Handle missing compression block size (#1385)
https://issues.apache.org/jira/browse/ORC-1357

Comment (abstractdog) [reply ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestOrcFile.java:1237]: yay, very cool, nice catch!
so we had previously made a completely invalid assertion without a code comment (without being aware that we're asserting based on an invalid behavior)

File: beeline/pom.xml [hunk 5]
```diff
@@ -84,6 +84,11 @@
       <groupId>jline</groupId>
       <artifactId>jline</artifactId>
     </dependency>
+    <dependency>
```
Comment (abstractdog) [inline beeline/pom.xml:87]: why is this needed?

Comment (zratkai) [reply beeline/pom.xml]: I think it was not intended. Removing.

**Comment by akshat0395** [comment]
Had a minor query which is covered by @abstractdog regarding hadoop-hdfs dependency in beeline pom.
Thanks for this change @zratkai 
LGTM, +1

File: beeline/pom.xml [hunk 6]
```diff
@@ -193,6 +193,11 @@
       <version>${powermock.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (akshat0395) [inline beeline/pom.xml:196]: @zratkai I see this dependency is again introduced in your last commit, did something change? do we need this? or this was just a commit miss?

**Review by abstractdog — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you everyone. It's a great milestone!
